### ISSUE TITLE
Lora Training Fails with TypeError: can only concatenate str (not "int") to str

### ISSUE THREAD

**Issue by xjdeng** [d8ahazard/sd_dreambooth_extension#875]
### Kindly read the entire form below and fill it out with the requested information.

**Please find the following lines in the console and paste them below. _If you do not provide this information, your
issue will be automatically closed._**

`
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0]
Commit hash: 91c8d0dcfc9a5d46aec47fd3eb34c633c104b5bb
Installing requirements for Web UI

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0]
Dreambooth revision: 9f4d931a319056c537d24669cb950d146d1537b0
SD-WebUI revision: 91c8d0dcfc9a5d46aec47fd3eb34c633c104b5bb

Checking Dreambooth requirements...
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.16rc425 installed.
[+] torch version 1.13.1+cu116 installed.
[+] torchvision version 0.14.1+cu116 installed.

#######################################################################################################
`

**Have you read the [Readme](https://github.com/d8ahazard/sd_dreambooth_extension#readme)?**
Yes

**Have you completely restarted the stable-diffusion-webUI, not just reloaded the UI?**
Yes

**Have you updated Dreambooth to the latest revision?**
Yes

**Have you updated the Stable-Diffusion-WebUI to the latest version?**
Yes

**No, really. Please save us both some trouble and update the SD-WebUI and Extension and restart before posting this.
Reply 'OK' Below to acknowledge that you did this.**

**Describe the bug**

I'm unable to train any LORA models; it crashes with the error "TypeError: can only concatenate str (not "int") to str".

Update: Dreambooth training seems to be working fine but I'm still unable to train LORA models.

**Provide logs**

If a crash has occurred, please provide the *entire* stack trace from the log, including the last few log messages *
before* the crash occurred.

Initializing dreambooth training...
Pre-processing 512: 100% 20/20 [00:00<00:00, 371.55it/s]
Concept requires 0 class images per instance image.
Sorting instance images: 100% 1/1 [00:00<00:00, 28339.89it/s]
Nothing to generate.
Replace CrossAttention.forward to use xformers
Found 0 reg images.
Preparing dataset...
Init dataset!
Preparing Dataset (With Caching)
Bucket 0 (512, 512, 0) - Instance Images: 20 | Class Images: 0 | Max Examples/batch: 20
Total Buckets 1 - Instance Images: 20 | Class Images: 0 | Max Examples/batch: 20
Caching latents...: 100% 20/20 [00:10<00:00,  7.42it/s]
Total images / batch: 20, total examples: 20
Caching latents...: 100% 20/20 [00:10<00:00,  1.92it/s]
Total dataset length (steps): 20
Initializing bucket counter!
  ***** Running training *****
  Num batches each epoch = 20
  Num Epochs = 100
  Batch Size Per Device = 1
  Gradient Accumulation steps = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Text Encoder Epochs: 100
  Total optimization steps = 2000
  Total training steps = 2000
  Resuming from checkpoint: False
  First resume epoch: 0
  First resume step: 0
  Lora: True, Adam: False, Prec: fp16
  Gradient Checkpointing: True
  EMA: False
  UNET: True
  Freeze CLIP Normalization Layers: False
  LR: 0.0001
  LoRA Text Encoder LR: 0.0001
  V2: False
Steps:   0% 0/2000 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/content/drive/MyDrive/ML/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/dreambooth.py", line 455, in start_training
    result = main(config, use_txt2img=use_txt2img)
  File "/content/drive/MyDrive/ML/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 989, in main
    return inner_loop()
  File "/content/drive/MyDrive/ML/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/memory.py", line 121, in decorator
    return function(batch_size, grad_size, prof, log_file, *args, **kwargs)
  File "/content/drive/MyDrive/ML/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 903, in inner_loop
    args.revision += train_batch_size
TypeError: can only concatenate str (not "int") to str
Steps:   0% 0/2000 [00:07<?, ?it/s]
Restored system models.

**Environment**

What OS? Google Colab

If Windows - WSL or native?

What GPU are you using? Tesla T4

**Screenshots/Config**
If the issue is specific to an error while training, please provide a
 screenshot of training parameters or the
db_config.json file from /models/dreambooth/MODELNAME/db_config.json
[db_config.zip](https://github.com/d8ahazard/sd_dreambooth_extension/files/10527931/db_config.zip)

**Comment by ArrowM** [comment]
Can you attach your db_config?

**Comment by xjdeng** [comment]
[db_config.zip](https://github.com/d8ahazard/sd_dreambooth_extension/files/10527933/db_config.zip)

**Comment by xjdeng** [comment]
> Does changing `args.revision += train_batch_size`
> 
> To
> 
> `args.revision += str(train_batch_size)`
> 
> Do anything?

I haven't tried that because I've been able to resolve this error and haven't seen it return (yet) by replacing my db_config.json with one from a model that trained fine and editing it accordingly.  It's worth mentioning that I did try modifying that line to:

`args.revision = int(args.revision)`

`args.revision += str(train_batch_size)`

But I ended up with more errors downstream.  I guess args.revision is supposed to be a str not an int by default at least that's what I infer from this.
